### PR TITLE
[SPARK-29834][SQL]DESC DATABASE should look up catalog like v2 commands

### PR DESCRIPTION
**Pull request by fuwhu**
### What changes were proposed in this pull request?
Add DescribeTableStatement and make "DESC DATABASE" look up catalog like v2 commands.

### Why are the changes needed?
It's important to make all the commands have the same table resolution behavior, to avoid confusing end-users. 
Also to fix [SPARK-29834](https://issues.apache.org/jira/browse/SPARK-29834).

### Does this PR introduce any user-facing change?
yes. When running DESC DATABASE, Spark fails the command if the current catalog is set to a v2 catalog, or the database name specified a v2 catalog.

### How was this patch tested?
New unit test

### CODE REVIEW

**Comment by wangyum** [comment]
ok to test

**Comment by wangyum** [comment]
retest this please

**Comment by wangyum** [comment]
cc @cloud-fan

File: sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4 [hunk 1]
```diff
@@ -199,7 +199,7 @@ statement
     | SHOW CREATE TABLE multipartIdentifier                            #showCreateTable
     | SHOW CURRENT NAMESPACE                                           #showCurrentNamespace
     | (DESC | DESCRIBE) FUNCTION EXTENDED? describeFuncName            #describeFunction
-    | (DESC | DESCRIBE) database EXTENDED? db=errorCapturingIdentifier #describeDatabase
+    | (DESC | DESCRIBE) database EXTENDED? db=multipartIdentifier      #describeDatabase
```
Comment (cloud-fan) [inline sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: let's be consistent with `CREATE (database | NAMESPACE)`, and also use `(database | NAMESPACE)` here.

Comment (cloud-fan) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: also remove `db=` as it's a namespace name now.

Comment (cloud-fan) [reply sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4]: also update the label `#describeDatabase` to `#describeNamespace`

File: sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala [hunk 2]
```diff
@@ -169,6 +169,10 @@ class ResolveSessionCatalog(
         DeleteFromTable(aliased, condition)
       }
 
+    case DescribeDatabaseStatement(databaseName, extended) =>
```
Comment (cloud-fan) [inline sql/core/src/main/scala/org/apache/spark/sql/catalyst/analysis/ResolveSessionCatalog.scala]: We can follow how `DropNamespaceStatement` is handled.

**Comment by fuwhu** [comment]
code messed up, open another PR #26513 to follow up

**Pull request closed without merging**